packages/e2e/cypress/integration/common/shared.ts

```diff
@@ -324,9 +324,29 @@ When('I mock {string} event', (eventName: string) => {
     throw new Error('Hub is not available on the window.');
   }
 
-  Hub.dispatch('auth', { event: eventName });
+  Hub.dispatch('auth', { event: eventName, data: {} });
 });
 
+When(
+  'I mock {string} event with fixture {string}',
+  async (eventName: string, fixture: string) => {
+    if (!window) {
+      throw new Error('window has not been set in the Cypress tests');
+    }
+
+    const Hub = window['Hub'];
+    if (!Hub) {
+      throw new Error('Hub is not available on the window.');
+    }
+
+    cy.wait(300)
```

Comment:
Why 300ms?

Reply:
This is fired whenever we need to mock the hub event to complete an auto sign in. The wait is added to ensure that the previous step completed. 

For example, in a real application, when the user clicks "confirm password", xState sends the confirm password to the backend, then it transitions to the "autoSignIn" state. Once the backend receives the confirm password the JS library triggers the "autoSignIn" hub event which xState receives and transitions to the "authenticated" state.

In our tests, since we are mocking out all the backend calls, the call to the backend is ignored for confirm password, and the "autoSignIn" hub event is fired immediately after the automation clicks the confirm password. This is before xState is in the `autoSignIn` state so it always fails. To get around this, I added a slight delay of 300ms so the xState machine will be in the correct state before we send the hub event.

Reply:
I'm slightly concerned about this race condition. Do you think we can adjust our xstate machine to immediately start listening for `autoSignIn` after confirmSignUp? I can help with xstate change if needed!

Reply:
I'm slightly concerned about this race condition. Do you think we can adjust our xstate machine to immediately start listening for autoSignIn after confirmSignUp? I can help with xstate change if needed!

Reply:
Update, synced offline, and I'll open a sub PR to resolve this in xstate.